eventGenerator/temperature.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+type TemperatureEvent struct {
+	Degree  float32
+	Name    string
+	Created time.Time
+	Event   map[string]string
+}
+
+// type Status struct {
+// 	Status int
+// }
+
+func toFixed(num float64, precision int) float64 {
+	output := math.Pow(10, float64(precision))
+	return float64(round(num*output)) / output
+}
+
+func round(num float64) int {
+	return int(num + math.Copysign(0.5, num))
+}
+
+func GenerateTemperatureEvent() {
```

Comment:
Please use generator, which we have in main;
Do NOT use an infinite loop for such things

Reply:
I'd add if you need to derive next value from a previous one - store that previous value in the upper scope